src/lightning/pytorch/trainer/supporters.py

```diff
@@ -125,74 +122,44 @@ def __next__(self) -> Any:
         finally:
             self.counter += 1
 
-    def __len__(self) -> Union[int, float]:
+    def __len__(self) -> int:
         return self.length
 
 
-class CombinedDataset:
-    """Combine multiple datasets and compute their statistics."""
+class _CombinationMode(TypedDict):
+    name: str
+    fn: Callable[[Union[int, float], ...], Union[int, float]]
+    default: Union[int, float]
 
-    COMPUTE_FUNCS = {"min_size": min, "max_size_cycle": max}
 
-    def __init__(self, datasets: Union[Sequence, Mapping], mode: str = "min_size"):
+_supported_modes = {
+    "min_size": _CombinationMode(name="min_size", fn=min, default=float("inf")),
+    "max_size_cycle": _CombinationMode(name="max_size_cycle", fn=max, default=float("-inf")),
+}
```

Comment:
I plan to add

```py
    "sequential": _CombinationMode(name="sequential", fn=int.__add__, default=0),
```

for sequential support with evaluation dataloaders

Reply:
Out of curiosity, where do you plan that these modes be specified? Still in the trainer constructor?

Reply:
Probably from requiring manual instantiation of `CombinedLoader`